ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -213,11 +357,67 @@ gcp_object_list_free(
     return 0;
 }
 
-int
-wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
+static int
+gcp_terminate(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session)
 {
-    WT_UNUSED(connection);
-    WT_UNUSED(config);
+    gcp_store *gcp = reinterpret_cast<gcp_store *>(storage_source);
+
+    if (--gcp->reference_count != 0)
+        return 0;
+
+    /*
+     * Terminate any active filesystems. There are no references to the storage source, so it is
+     * safe to walk the active filesystem list without a lock. The removal from the list happens
+     * under a lock. Also, removal happens from the front and addition at the end, so we are safe.
+     */
+    while (!gcp->fs_list.empty()) {
+        gcp_file_system *fs = gcp->fs_list.front();
+        gcp_file_system_terminate(&fs->file_system, session);
+    }
+
+    std::cout << "gcp_terminate: Terminated GCP storage source." << std::endl;
```

Comment:
I suggest using `terminated` to keep capitalization consistent with the other messages.